sponsorship/src/main/java/com/kyriba/conference/sponsorship/dao/SponsorRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.kyriba.conference.sponsorship.dao;
+
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Repository;
+
+import java.util.stream.StreamSupport;
+
+
+/**
+ * @author Aliaksandr Samal
+ */
+@Repository
+public interface SponsorRepository extends CrudRepository<Sponsor, Long>
+{
+  @Nullable
+  default Sponsor readByEmail(String email)

```

Comment:
Declare method with valid Spring Data syntax and don't read all sponsors for filtering.